### PR TITLE
feat:  switch to bun and elysia

### PR DESCRIPTION
**Pull request by KaiVolland**
This transforms the rest backend by using a diffrent tech-stack.

It is now built on [Bun](https://bun.sh) and [ElysiaJS](https://elysiajs.com/)

BREAKING CHANGE: switch to bun and elysia

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1 +1,7 @@
-node_modules/
\ No newline at end of file
+node_modules
+.github
+.gitignore
+Dockerfile
+LICENSE
```
Comment (marcjansen) [inline .dockerignore:5]: I think this fine when copied over, isn't it?

File: package.json [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
+  "main": "src/index.ts",
   "scripts": {
-    "lint": "eslint index.js",
-    "start-dev": "nodemon index.js",
-    "start": "node index.js",
-    "test": "NODE_API_PORT=9999 npm run lint && mocha test/test.spec.js --exit"
+    "test": "bun test",
+    "dev": "bun run --watch src/index.ts",
+    "start": "bun run src/index.ts",
+    "lint": "echo \"Linting not configured\" && exit 0"
```
Comment (marcjansen) [inline package.json:11]: This is a step backwards, or am I not understanding things?

File: src/routes/api.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+      throw new ParseError('Error reading input', readResponse?.errors?.[0]?.message);
+    }
+  }
+
+  // if no targetParser is given we return a geostyler-style
+  if (targetParser === undefined) {
+    return readResponse.output;
+  }
+
+
```
Comment (marcjansen) [inline src/routes/api.ts:109]:
```suggestion
```

File: src/routes/info.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  request
+}) => {
+  if (request.headers.get('accept') === 'application/json') {
+    return {
+      'geostyler-rest': version,
+      'geostyler-mapbox-parser': getVersionString('geostyler-mapbox-parser'),
+      'geostyler-mapfile-parser': getVersionString('geostyler-mapfile-parser'),
+      'geostyler-qgis-parser': getVersionString('geostyler-qgis-parser'),
+      'geostyler-sld-parser': getVersionString('geostyler-sld-parser'),
+      'geostyler-lyrx-parser': getVersionString('geostyler-lyrx-parser')
```
Comment (marcjansen) [inline src/routes/info.tsx:25]: A stylistic thing: I would have created the object outside of the `if`, and used it both branches (returned or used to construct HTML)

File: src/routes/info.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+              GeoStyler ArcGIS Parser: {getVersionString('geostyler-lyrx-parser')}
+            </li>
+          </ul>
+        </body>
+      </html>
+    );
+  }
+};
+
+const getVersionString = (parserName: string) => {
```
Comment (marcjansen) [inline src/routes/info.tsx:58]: Should be moved to the top.

**Comment by KaiVolland** [comment]
Thanks for the review @marcjansen. I added some commits to adapt to your remarks.

**Review by marcjansen — APPROVED** [review]
Thanks looks good to me.

File: src/routes/api.ts [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+  })
+};
+
+export const transform: Handler = async ({
+  body,
+  query: { sourceFormat, targetFormat }
+}) => {
+
+  if (!body) {
+    log.error('Error: No source style style given in POST body.');
```
Comment (jansule) [inline src/routes/api.ts:77]:
```suggestion
    log.error('Error: No source style given in POST body.');
```

File: src/routes/api.ts [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+    throw new ParseError('Error', 'No source style style given in POST body.');
```
Comment (jansule) [inline src/routes/api.ts:78]:
```suggestion
    throw new ParseError('Error', 'No source style given in POST body.');
```

File: tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,106 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig to read more about this file */
```
Comment (jansule) [inline tsconfig.json:3]: Do we want to keep all of the commented code?

**Review by jansule — COMMENTED** [review]
Nice work. Should the change from `/geostyler-rest/api-docs` to `/api-docs` be included in a breaking change commit message?

File: tsconfig.json (hunk 8 above)
Comment (KaiVolland) [reply tsconfig.json:3]: This file is generated so i'd just keep it as is.

**Comment by KaiVolland** [comment]
I'll squash and merge this PR anyway so the BREAKING CHANGE will be documented.

**Review by jansule — APPROVED** [review]

**Pull request merged**